devito/compiler.py

```diff
@@ -366,9 +367,22 @@ def jit_compile(soname, code, compiler):
     target = str(get_jit_dir().joinpath(soname))
     src_file = "%s.%s" % (target, compiler.src_ext)
 
-    # This makes a suite of cache directories based on the soname
-    cache_dir = get_codepy_dir().joinpath(soname[:7])
-    cache_dir.mkdir(parents=True, exist_ok=True)
+    if configuration['jit-backdoor'] is False:
+        # Typically we end up here
+        # Make a suite of cache directories based on the soname
+        cache_dir = get_codepy_dir().joinpath(soname[:7])
+        cache_dir.mkdir(parents=True, exist_ok=True)
+    else:
+        # Warning: dropping `code` on the floor in favor to whatever is written
+        # within `src_file`
+        try:
+            with open(src_file, 'r') as f:
+                code = f.read()
+            # Bypass the devito JIT cache
+            cache_dir = mkdtemp()
+        except FileNotFoundError:
```

Comment:
Fall back to deefault jit instead?

Reply:
no, I prefer to error out. The user tried to bypass the jitter, but something went south, and this should be notified immediately so that one can figure out what's going on -- running the other code path could actually create confusion !